src/iaux-underlined-tab-bar.ts

```diff
@@ -0,0 +1,195 @@
+import {
+  html,
+  css,
+  LitElement,
+  property,
+  customElement,
+  query,
+  queryAll,
+  PropertyValues,
+  internalProperty,
+  CSSResult,
+} from 'lit-element';
+import { nothing } from 'lit-html';
+import './iaux-underlined-tab-bar-loading-dots';
+
+export interface TopTab {
+  displayName: string;
+  tabTitle?: string;
+}
+
+@customElement('iaux-underlined-tab-bar')
+export class UnderlinedTabBar extends LitElement {
+  @property({ type: Array }) entries: TopTab[] = [];
+
+  @property({ type: Number }) selectedIndex = 0;
+
+  @property({ type: Number }) widthMultiplier = 0.8;
+
+  @property({ type: Boolean }) isLoading = false;
+
+  @internalProperty() animatedUnderline = false;
+
+  @query('.horizontalScroll') container!: HTMLDivElement;
+
+  @queryAll('button') listOfButtons!: HTMLButtonElement[];
+
+  private itemClicked(e: Event, index: number) {
+    const event = new CustomEvent('itemclicked', {
+      detail: {
+        index,
+      },
+    });
+    const indexDistance = Math.abs(this.selectedIndex - index);
+    const animationTime = Math.min(0.25 * indexDistance, 0.5);
```

Comment:
suggestion: make `0.5` a settable parameter (or a constant) with a name that communicates its purpose, maybe something like `ANIMATION_SECONDS` (if a constant)?

Reply:
Thanks, added new properties to customize animation duration.